go.mod

```diff
@@ -1,4 +1,4 @@
-module github.com/DataDog/temporalite
+module github.com/temporalio/temporalite
```

Comment:
Do we actually want to use a vanity url here eg. `go.temporal.io/temporalite`?

Reply:
After off-PR discussion, while we'll confirm with a larger audience internally, this is good to start with instead of a vanity URL

Reply:
Discussed in slack -- we'll go ahead and merge as-is for now. We may still update the module name again in the future.